bin/prover-client/src/service/task.rs

```diff
@@ -0,0 +1,61 @@
+//! ProofTask type for Prover Service integration
+//!
+//! This module defines the ProofTask newtype wrapper around ProofContext
+//! and implements the ProgramType trait for registry-based dispatch.
+
+use serde::{Deserialize, Serialize};
+use strata_paas::ProgramType;
+use strata_primitives::proof::ProofContext;
+
+/// Routing key for ProofContext - used for dynamic dispatch in Prover Service registry
+///
+/// This enum represents the different variants of ProofContext without carrying
+/// the actual data, allowing the registry to route requests to the correct handler.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum ProofContextVariant {
+    EvmEeStf,
+    ClStf,
+    Checkpoint,
+}
```

Comment:
The name `ProofContextVariant` feels pretty obtuse to me, but the doc calls this a "routing key".  I get the idea we're trying to capture here but could we pick something more precise?